framework/include/utils/CompileTimeDerivatives.h

```diff
@@ -527,6 +583,83 @@ class CTDiv : public CTBinary<L, R>
   using CTBinary<L, R>::_right;
 };
 
+enum class CTComparisonEnum
+{
+  Less,
+  LessEqual,
+  Greater,
+  GreaterEqual,
+  Equal,
+  Unequal
+};
+
+/**
+ * Binary comparison operator node
+ */
+template <CTComparisonEnum C, typename L, typename R>
+class CTCompare : public CTBinary<L, R>
+{
+public:
+  CTCompare(L left, R right) : CTBinary<L, R>(left, right) {}
```

Comment:
Is pass-by-value vs. pass-by-reference paradigm relevant in these compile time derivatives?

Reply:
Hmmmmmm. That's a good question. This would only matter at construction and we'd be passing temporary objects almost every time. My guess is that the compiler should be able to optimize this very well. But I can try and experiment with CompilerExplorer to see if it makes a difference.

Reply:
I think it does not matter. Everything that's required to build the type is known at compile time. Granted this is only a very cursory test, but it looks like compilers optimize this very well: https://www.godbolt.org/z/bd168vj6W